test cases/common/267 dep as_static/meson.build

```diff
@@ -0,0 +1,31 @@
+project(
+    'as_static',
+    ['c'],
+    meson_version: '>= 1.3.0',
+)
+
+lib1 = library('lib1', 'lib1.c', vs_module_defs: 'lib1.def')
+dep1 = declare_dependency(link_with: lib1)
+
+lib2 = both_libraries('lib2', 'lib2.c', vs_module_defs: 'lib2.def')
+dep2 = declare_dependency(link_with: lib2)
+
+lib3 = shared_library('lib3', 'lib3.c', vs_module_defs: 'lib3.def')
+dep3 = declare_dependency(link_with: lib3)
+
+lib4 = static_library('lib4', 'lib4.c')
+dep4 = declare_dependency(link_with: lib4)
+
+
+dep_default = declare_dependency(dependencies: [dep1, dep2, dep3, dep4])
+main_default = executable('main_default', 'main.c', dependencies: [dep_default])
+test('default', main_default)
+
+dep_static = declare_dependency(dependencies: [dep1, dep2, dep3, dep4]).as_static(recursive: true)
+main_static = executable('main_static', 'main.c', dependencies: [dep_static])
+test('static', main_static)
+
+
+# FIXME: this doesn't really test that expected lib versions are linked,
+#        but I don't know how I could test this...
+#        Maybe should I write a unit test for that?
```

Comment:
Maybe you can merge this in `unit/113 complex link cases` that already has a unit test in `unittests/linuxliketests.py` to check how ninja links those libs.

Reply:
I don't think you have to worry too much about Windows for this kind of test. Your PR is platform agnostic.

Reply:
For reference, I'm waiting for #11981 to write the test